main.rb

```diff
@@ -0,0 +1,120 @@
+module Enumerable
+  # each
+  def my_each
+    each do |i|
+      yield i
+    end
+  end
+
+  # each_with_index
+  def my_each_with_index
+    result = map do |element, index|
+      yield "#{index}: #{element} "
+    end
+    result
+  end
+
+  # select
+  def my_select
+    result = []
+    each do |i|
+      result << i if yield(i)
+    end
+    result
+  end
```

Comment:
- [ ]  `my_select` should returns an enumerator if no block is given.
`p [1, 2, 3].my_select` returns `#<Enumerator>`